src/algorithms/toolbox.jl

```diff
@@ -160,39 +160,46 @@ function MPSKit.transfer_spectrum(
 end
 
 """
-    correlation_length(peps::InfinitePEPS, env::CTMRGEnv; num_vals=2)
+Adjoint of an MPS tensor, but permutes the physical spaces back into the codomain.
+Intuitively, this conjugates a tensor and then reinterprets its 'direction' as an MPS
+tensor.
+"""
+function _dag(A::MPSKit.GenericMPSTensor{S,N}) where {S,N}
```

Comment:
Is this to avoid the fact that MPSKit always daggers the bottom MPS? It feels like a hack, but I guess there's not really an alternative?

Reply:
Works for me, just wanted to check if I got that right.